### PR TITLE
Add stderr routing for printed errors; clean up color handling; add pprint hooks for compile-time messages

### PR DESCRIPTION
**Pull request by qcjames53**
Hi! I refactored a lot in this PR but with the purpose of keeping existing functionality as similar as possible to the current JMC tool. Please let me know if any of the following changes went too far and I'd be happy to revert changes or meet in the middle.

#### Changes made to printing: 
- `pprint` is used wherever possible throughout JMC.
- Added a message hook for the `compile` module so that it can use `pprint`. From what I can tell this is standard practice for other Python modules. This can be extended in the future if `compile` needs to pass more info back to the `terminal` module.
- Added `eprint` for error messages. By default this prints to stderr.
- Color data should now always be cleaned up on exit.
- ANSI escape codes are no longer emitted in non-TTY contexts.
- Removed black text backgrounds.
- Added `os.system("")` to main function so that interactive and non-interactive modes both have color on Windows 10 terminal.
- Cleaned up terminal module imports to all be top-level. Left compile module imports alone since this is out of scope.
- Overall grammar pass ("Configuration file does not exists." -> "Configuration file does not exist.")
- Added unit tests for the major additions + corrected edge cases. Most changes remain untested since terminal color isn't really that big of a deal :)
- Unstyled `Colors.NONE` for better readability in light terminals.

### CODE REVIEW

File: src/jmc/terminal/utils.py [hunk 1]
```diff
@@ -8,38 +9,55 @@
...
+
+
+def pprint(values, color: Colors = Colors.NONE, file=sys.stdout):
     """
-    Print but with colors
+    Print colored text
 
-    :param values: Value for printing
-    :param color: color of printing
+    :param: values: Output text
```
Comment (WingedSeal) [inline src/jmc/terminal/utils.py]: There's a typo, extra `:` after `param` shouldn't be there

File: src/tests/unit/test_terminal.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+from jmc.terminal.utils import Colors, eprint, pprint
+
+
+class TestHooks(unittest.TestCase):
+    def test_default_handler_is_print(self):
+        from jmc.compile import hooks
+        self.assertIs(hooks._message_handler, print)
+
+    def test_register_replaces_handler(self):
+        received = []
```
Comment (WingedSeal) [inline src/tests/unit/test_terminal.py]: Can you add type hint here? `received: list[...] =`

File: src/tests/unit/test_terminal.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        register_message(received.append)
+        emit_message("hello")
+        self.assertEqual(received, ["hello"])
+
+    def test_emit_calls_through_handler(self):
+        received = []
```
Comment (WingedSeal) [inline src/tests/unit/test_terminal.py]: Here as well

**Review by WingedSeal — CHANGES_REQUESTED** [review]

File: src/jmc/terminal/utils.py (hunk 1 above)
Comment (qcjames53) [reply src/jmc/terminal/utils.py]: Resolved, thanks!

File: src/tests/unit/test_terminal.py (hunk 2 above)
Comment (qcjames53) [reply src/tests/unit/test_terminal.py]: Both should be fixed.

**Comment by qcjames53** [comment]
Thank you so much for the review. I fixed everything brought up.

Sorry about the "fixed this broken thing that was very messy" comments in the PR description. I was pretty tired by the time I submitted this and didn't realize how it came across. JMC is a pretty awesome tool and I don't want to criticize something you've developed and maintained yourself for years. That's all.
